presto-main/src/main/java/io/prestosql/operator/HashAggregationOperator.java

```diff
@@ -510,6 +522,10 @@ private void closeAggregationBuilder()
             // aggregationBuilder.close() will release all memory reserved in memory accounting.
             // The reference must be set to null afterwards to avoid unaccounted memory.
             aggregationBuilder = null;
+            if (aggregationBuilderMemoryContext != null) {
+                aggregationBuilderMemoryContext.setBytes(0);
```

Comment:
You're zeroing memory contexts at the end of this method.
Either add `operatorContext.localSystemMemoryContext().setBytes(0);` there (instead of this line), or remove the all other `.setBytes(0)` calls there. Or explain :)

Reply:
True. Didn't notice that.
Just `memoryContext.setBytes(0);` at end of this method is enough